### DISCUSSION TITLE
location of installed package

### DISCUSSION THREAD

**Discussion by btcwid** [pnpm/pnpm#3253] [Q&A]
I joined a project that in the meantime lost the dev that set things up.

Working in a monorepo, one package  ("client") is depending on another package also from the monorepo ("dep").  Packages scoped by "org-scope".

New devs join, working in "client", have a package.json with "dep" in the usual dependencies object location.

`pnpm i` installs it OK, , but `pnpm view` gives @org-scope/dep@latest' is not in the npm registry.

Nothing interesting AFAICT in project or user `.npmrc` 

Trying to understand, **where is pnpm getting this package from**, what commands can help?

**Comment by zkochan** [comment]
Is `@org-scope/dep` a private package? Do you want to search for `@org-scope/dep` in the monorepo?

**Reply by btcwid** [reply]
Thank you for those prompts.

Yes, it has `"private": true` in `package.json`.

Searching ...

`@org-scope/dep` shows up in: 
`pnpm-lock.yaml` of client parent dir, with `link:<relpath-to-dep>`
in `dependencies` object of `package.json` of client
as `name`, in `package.json` of dep
& other less interesting-seeming places

-> so, earlier team member did `pnpm link <relpath-to-dep>` in root?

side note:
I thought the `view` subcommand would output provenance regardless of source - in this case, it would show the link info from the lockfile - but I don't know where I got that idea.  Also I see that this command is not actually supported in `pnpm`, so, I'm making up stuff on top of making up stuff.

**Reply by zkochan** [reply]
you can try `pnpm ls -r <depname>`
